rita_exit/src/database/database_tools.rs

```diff
@@ -637,15 +635,19 @@ pub fn get_client_subnet(
 fn generate_iterative_client_subnet(
     exit_sub: IpNetwork,
     ind: u64,
-    subprefix: usize,
+    subprefix: u8,
 ) -> Result<IpNetwork, RitaExitError> {
-    // Convert IpNetwork into IPAdress type
-    let network: IPAddress = IPAddress::parse(exit_sub.to_string())
-        .expect("Paniced while parsing the exit subnet: IpNetwork -> IPAddress");
-    let mut net = network.network();
-    net.prefix = net.prefix.from(subprefix).unwrap();
+    let net;
+    let net_as_int: u128 = if let IpAddr::V6(addr) = exit_sub.network() {
+        net = Ipv6Network::new(addr, subprefix).unwrap();
```

Comment:
a comment for what this logic is doing would be good. I can't quite follow it.

Reply:
fixed